runtime/binding-http/src/main/java/io/aklivity/zilla/runtime/binding/http/internal/config/HttpBindingConfig.java

```diff
@@ -164,6 +195,225 @@ private Function<Function<String, String>, String> asAccessor(
+    }
+
+    public HttpRequestType resolveRequest(
+        HttpBeginExFW beginEx)
+    {
+        HttpRequestType result = null;
+        if (requests != null && !requests.isEmpty())
+        {
+            String path = resolveHeaderValue(beginEx, HEADER_PATH);
+            String method = resolveHeaderValue(beginEx, HEADER_METHOD);
+            String contentType = resolveHeaderValue(beginEx, HEADER_CONTENT_TYPE);
+            for (HttpRequestType request : requests)
+            {
+                boolean isMatch = false;
+                isMatch |= method == null || request.method == null || method.equals(request.method.name());
+                isMatch |= contentType == null || request.contentType == null || request.contentType.contains(contentType);
+                isMatch &= parseParams(request, path);
+                if (isMatch)
+                {
+                    result = request;
+                    break;
+                }
+            }
+        }
+        return result;
+    }
+
+    private boolean parseParams(
+        HttpRequestType request,
+        String path)
+    {
+        boolean result = false;
+        String pattern = String.format("^%s/?(\\?(?<query0>.*))?$",
+            request.path.replaceAll("\\{([a-zA-Z_]+)\\}", "(?<$1>.+?)"));
+        Matcher matcher = Pattern.compile(pattern).matcher(path);
+        if (matcher.matches())
+        {
+            this.pathParams = parsePathParams(request.path, matcher);
+            this.queryParams = parseQueryParams(matcher.group("query0"));
+            result = true;
+        }
+        return result;
+    }
+
+    private Map<String, String8FW> parsePathParams(
+        String pathTemplate,
+        Matcher matcher)
+    {
+        Map<String, String8FW> result = new HashMap<>();
+        String[] segments = pathTemplate.split("/");
+        for (String segment : segments)
+        {
+            if (segment.startsWith("{") && segment.endsWith("}"))
+            {
+                String name = segment.substring(1, segment.length() - 1);
+                String8FW value = new String8FW(URLDecoder.decode(matcher.group(name), UTF_8));
+                result.put(name, value);
+            }
+        }
+        return result;
+    }
+
+    private Map<String, String8FW> parseQueryParams(
+        String query)
+    {
+        Map<String, String8FW> queryParams = new HashMap<>();
+        if (query != null && !query.isBlank())
+        {
+            String[] paramPairs = query.split("&");
+            for (String paramPair : paramPairs)
+            {
+                String[] keyValue = paramPair.split("=");
+                if (keyValue.length == 2)
+                {
+                    String key = URLDecoder.decode(keyValue[0], UTF_8);
+                    String8FW value = new String8FW(URLDecoder.decode(keyValue[1], UTF_8));
+                    queryParams.put(key, value);
+                }
+            }
+        }
+        return queryParams;
```

Comment:
There's a lot of object allocation going on in these parse methods. Let's aim to eliminate it.

Reply:
We should be able to simplify this loop to use a Pattern Matcher for each query parameter, calling `.find()` for each one, then extracting the parameter name and value from the matcher via named groups.